src/OrchardCore/OrchardCore.ContentManagement.Display/ContentDisplay/ContentItemDisplayCoordinator.cs

```diff
@@ -135,39 +143,64 @@ public async Task BuildDisplayAsync(ContentItem contentItem, BuildDisplayContext
 
                     if (part.GetType() == typeof(ContentPart) && partTypeName != contentTypePartDefinition.ContentTypeDefinition.Name)
                     {
+                        // Todo: var shapeType = partTypeName; for Placement fix PR #8935
                         var shapeType = context.DisplayType != "Detail" ? "ContentPart_" + context.DisplayType : "ContentPart";
 
                         var shapeResult = new ShapeResult(shapeType, ctx => ctx.ShapeFactory.CreateAsync(shapeType, () => new ValueTask<IShape>(new ZoneHolding(() => ctx.ShapeFactory.CreateAsync("Zone")))));
                         shapeResult.Differentiator(partName);
+                        shapeResult.Name(partName);
                         shapeResult.Location("Content");
 
-                        await shapeResult.ApplyAsync(context);
-
-                        var contentPartShape = shapeResult.Shape;
+                        shapeResult.Displaying(ctx =>
+                        {
+                            string[] displayTypes = new[] { "", "_" + ctx.Shape.Metadata.DisplayType };
 
-                        // Make the ContentPart name property available on the shape
-                        var dynamicContentPartShape = contentPartShape;
-                        dynamicContentPartShape.Properties[partTypeName] = part.Content;
-                        dynamicContentPartShape.Properties["ContentItem"] = part.ContentItem;
+                            foreach (var displayType in displayTypes)
+                            {
+                                // Fall back to default template of ContentPart, if there is not template for shape type(partTypeName)
+                                // eg. ContentPart, ContentPart.Summary
+                                ctx.Shape.Metadata.Alternates.Add($"ContentPart{displayType}");
```

Comment:
I still don't quite understand why we are adding an alternate for `ContentPart`, i.e. when the first element of `displayTypes` is empty, when in this case the shape name itself is already `ContentPart`, i.e from L147, and L149

Reply:
Line 147 will be removed and replaced with 146 in PR  #8935.  Anyway I am removing it from here for now